python/GangaAtlas/Lib/Athena/Athena.py

```diff
@@ -1154,13 +1154,25 @@ def prepare(self, force=False, **options):
         if self.atlas_exetype in ['EXE']: #and not self.athena_compile:  - for EXE, compilation decides what the tarball is called
             maxFileSize = config['EXE_MAXFILESIZE']
             archiveName, archiveFullName = create_tarball(self.userarea, runDir, currentDir, archiveDir, self.append_to_user_area, self.exclude_from_user_area, maxFileSize, self.useAthenaPackages, verbose, self.athena_compile )
-        else:
-            archiveName = ""
+
             if AthenaUtils.useCMake():
                 self.useCMake = True
-                archiveName,archiveFullName = AthenaUtils.archiveWithCpack(True,tmpDir,True)
+        else:
+            # compilation determines whether to send the sources across as well
+            archiveName = ""
+            if self.athena_compile:
+                if AthenaUtils.useCMake():
+                    self.useCMake = True
+                    archiveName,archiveFullName = AthenaUtils.archiveWithCpack(True,tmpDir,True)
+
+                archiveName, archiveFullName = AthenaUtils.archiveSourceFiles(self.userarea, runDir, currentDir, archiveDir, verbose, self.glue_packages, config['dereferenceSymLinks'], archiveName=archiveName)
```

Comment:
Doesn't this overload the `achiveName, archiveFullName` above if `not AthenaUtils.useCMake()`?
(This could be intended behaviour so apologies if it is :P )